Parse/Internal/Object/Subclassing/ObjectSubclassInfo.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Reflection;
+
+#if UNITY
+using TypeInfo = System.Type;
+#endif
+
+namespace Parse.Internal {
+  internal class ObjectSubclassInfo {
+    public ObjectSubclassInfo(Type type, ConstructorInfo constructor) {
+      TypeInfo = type.GetTypeInfo();
+      ClassName = GetClassName(TypeInfo);
+      Constructor = constructor;
+    }
+
+    public TypeInfo TypeInfo { get; private set; }
+    public String ClassName { get; private set; }
+    private ConstructorInfo Constructor { get; set; }
+
+    private Object mappingsMutex = new Object();
+    private Dictionary<String, String> propertyMappings;
+
+    public IDictionary<String, String> PropertyMappings {
+      get {
+        lock (mappingsMutex) {
+          propertyMappings = propertyMappings ?? TypeInfo.AsType().GetProperties()

```

Comment:
That makes sense. Let's put this inside initialization and drop the mutex. Every call to `[]`, `Set`, `Remove`, or even a simple `ObjectId` will call this props. So optimistically creating this this for every subclass will always be useful.